app/services/trainees/create_from_apply.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+module Trainees
+  class CreateFromApply
+    include ServicePattern
+
+    def initialize(application:)
+      @application = application
+    end
+
+    def call
+      trainee.save!
+      trainee
+    end
+
+  private
+
+    attr_reader :application
+
+    def attributes
+      @attributes ||= parsed_application["attributes"]
+    end
+
+    def parsed_application
+      @parsed_application ||= JSON.parse(application.application)
+    end
+
+    def trainee
+      @trainee ||= Trainee.new(mapped_attributes)
+    end
+
+    def mapped_attributes
+      {
+        apply_application: application,
+        provider: application.provider,
+        trainee_id: raw_trainee["id"],
+        first_names: raw_trainee["first_name"],
+        last_name: raw_trainee["last_name"],
+        date_of_birth: raw_trainee["date_of_birth"],
+        email: raw_contact_details["email"],
+        # TODO: Use actual route type, if found.
+        training_route: 1,
+        degrees: degrees,
+      }.merge(address)
+    end
+
+    def degrees
+      raw_degrees.map { |degree| ::Degrees::CreateFromApply.call(attributes: degree) }
+    end
+
+    def address
+      raw_contact_details["country"] != "GB" ? international_address : uk_address
```

Comment:
What about Northern Ireland?

Reply:
I believe this should be okay as apply are using ISO 3166-2 :gb: 
 https://github.com/DFE-Digital/apply-for-teacher-training/blob/ca6235e6cd8449b5f24904b90449e8598c937828/config/initializers/countries.rb#L189

Reply:
phew!